vpp/clientconfigsrv.go

```diff
@@ -0,0 +1,117 @@
+package vpp
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/satori/go.uuid"
+	"strings"
+)
+
+type ClientContext struct {
+	HostName string `json:"hostname"`
+	Guid     string `json:"guid"`
+}
+
+type Location struct {
+	LocationName string `json:"locationName"`
+	LocationId   int    `json:"locationId"`
+}
+
+type ClientConfigSrvResponse struct {
+	ClientContext      interface{} `json:"clientContext"`
+	AppleId            string      `json:"appleId"`
+	OrganizationIdHash string      `json:"organizationIdHash"`
+	Status             int         `json:"status"`
+	OrganizationId     int         `json:"organizationId"`
+	UId                string      `json:"uId"`
+	CountryCode        string      `json:"countryCode"`
+	Location           Location    `json:"location"`
+	ApnToken           string      `json:"apnToken"`
+	Email              string      `json:"email"`
+}
+
+type ClientConfigSrv struct {
+	ClientContext      ClientContext `json:"clientContext"`
+	AppleId            string        `json:"appleId"`
+	OrganizationIdHash string        `json:"organizationIdHash"`
+	Status             int           `json:"status"`
+	OrganizationId     int           `json:"organizationId"`
+	UId                string        `json:"uId"`
+	CountryCode        string        `json:"countryCode"`
+	Location           Location      `json:"location"`
+	ApnToken           string        `json:"apnToken"`
+	Email              string        `json:"email"`
+}
+
+func (c *Client) GetClientConfigSrv() (*ClientConfigSrv, error) {
+	request := map[string]interface{}{
```

Comment:
create request and response structs for each action. 
If you want to keep the request as an anonymous type/variable thats ok. See here as an example https://github.com/micromdm/micromdm/blob/a230831725f173f779cf7e030a93a71b32e8ab13/dep/devices.go#L108-L112

Reply:
I believe I have address these in the commit that I just pushed